### PR TITLE
Update knight valid move

### PR DESCRIPTION
**Pull request by Edwin-F**
I updated the knight valid move method with suggestions about using super. I also updated rspec tests for out_of boundary? method nested within the valid_move? method. 
Note that knight model valid_move? does not include the is_obstructed? method since it can hop over its own pieces.

### CODE REVIEW

File: app/models/piece.rb [hunk 1]
```diff
@@ -35,8 +35,6 @@ def check_mate?
   def valid_move?(new_row, new_column)
     return false if out_of_boundary?(new_row, new_column)
     return false if no_move?(new_row, new_column)
-    #return false unless proper_length?(new_row, new_column)
-    true
```
Comment (smiths15) [inline app/models/piece.rb]: @Ruby-Hot-Chili-Peppers/MightyPawn should "true" be here or in the specific piece model. I only bring this up because I think it was the opposite in the pawn_move branch. Thoughts?

File: app/models/knight.rb [hunk 2]
```diff
@@ -4,9 +4,13 @@ def symbol
   end
 
    #should return false if move is greater than 1 space
```
Comment (smiths15) [inline app/models/knight.rb]: remove comment

File: app/models/knight.rb [hunk 3]
```diff
@@ -4,9 +4,13 @@ def symbol
...
-  def proper_length?(new_row, new_column)
+
+
+  def valid_move?(new_row, new_column)
+    return false if super == false
     delta_row = (new_row - position_row).abs
     delta_col = (new_column - position_column).abs
-    ((delta_row == 1) && (delta_col == 2)) || ((delta_row == 2) && (delta_col == 1))
+    return false unless ((delta_row == 1) && (delta_col == 2)) || ((delta_row == 2) && (delta_col == 1))
```
Comment (smiths15) [inline app/models/knight.rb:10]: looks great! 👍

File: spec/models/knight_spec.rb [hunk 4]
```diff
@@ -83,11 +48,21 @@
           expect(@knight.valid_move?(1,0)).to be true
         end
 
-        it 'returns false for invalid move' do
+        it 'returns false if out of bounds for new_row' do
```
Comment (smiths15) [inline spec/models/knight_spec.rb:65]: change wording "returns true if out of bounds for new_row"

File: spec/models/knight_spec.rb [hunk 5]
```diff
@@ -83,11 +48,21 @@
...
-        it 'returns false for invalid move' do
+        it 'returns false if out of bounds for new_row' do
           #2 down then left and out of bounds
+          expect(@knight.out_of_boundary?(-2,0)).to be true
           expect(@knight.valid_move?(-2,0)).to be false
-          #2 down then right and out of bounds
-          expect(@knight.valid_move?(-2,2)).to be false
+        end
+
+        it 'returns false if out of bounds for new_column' do
```
Comment (smiths15) [inline spec/models/knight_spec.rb:70]: change wording "returns true if out of bounds for new_column"

**Review by smiths15 — CHANGES_REQUESTED** [review]
Looks great overall. See comments about changes

File: app/models/piece.rb (hunk 1 above)
Comment (xelzi) [reply app/models/piece.rb]: Good question, I was thinking the same thing. I don't think any piece will just call valid_move? from the piece file. Thus, I agree that we can probably get rid of true here because the super valid_move? will only tell us if a move is wrong, not if it is completely correct.

Comment (Edwin-F) [reply app/models/piece.rb]: I agree. The super valid_move? already has return true statement, and not having it in the valid_move? in piece can make the code a bit DRYer.

File: spec/models/knight_spec.rb [hunk 6]
```diff
@@ -19,6 +19,20 @@
         end
       end
 
+      context 'out_of_boundary?' do
```
Comment (smiths15) [inline spec/models/knight_spec.rb]: looks great! 👍

File: app/models/piece.rb [hunk 7]
```diff
@@ -63,7 +63,7 @@ def is_obstructed?(row_final, col_final)
       return Piece.exists?(position_row: row_range, position_column: col_init)
 
       #diagonal case
-    elsif ((row_final - row_init).to_f/(col_final - col_init).to_f ).abs == 1
+    elsif (row_final - row_init).abs == (col_final - col_init).abs
```
Comment (smiths15) [inline app/models/piece.rb:66]: looks great! 👍

**Review by smiths15 — APPROVED** [review]

**Review by xelzi — APPROVED** [review]
